samples/excel/89-preview-apis/events-disable-events.yaml

```diff
@@ -0,0 +1,166 @@
+id: excel-events-disable-events
+name: Disable Events
+description: Toggle event firing on and off for performance.
+host: EXCEL
+api_set:
+    ExcelAPI: 1.8
+script:
+    content: |+
+        $("#toggleEvents").click(() => tryCatch(toggleEvents));
+        $("#setup").click(() => tryCatch(setup));
+        $("#refreshData").click(() => tryCatch(addOrRefreshData));
+        $("#registerSumChangedHandlers").click(() => tryCatch(registerSumChangedHandlers));
+
+        async function toggleEvents() {
+            await Excel.run(async (context) => {
+                context.runtime.load("enableEvents");
+                await context.sync();
+                const eventBoolean = !context.runtime.enableEvents
+                context.runtime.enableEvents = eventBoolean;
+                if (eventBoolean) {
+                    console.log("Events are currently on.");
+                } else {
+                    console.log("Events are currently off.");
+                }
+                await context.sync();
+            });
+        }
+
+        async function registerSumChangedHandlers() {
+            await Excel.run(async (context) => {
+                const sheet = context.workbook.worksheets.getActiveWorksheet();
+                const sumRange = sheet.getRange("B20:P20");
+                sumRange.load("columnCount");
+                await context.sync();
+
+                for (var i = 0; i < sumRange.columnCount; i++) {
+                    let sumBinding = context.workbook.bindings.add(sumRange.getCell(0,i), Excel.BindingType.range, "SumBinding" + i);
+                    sumBinding.onDataChanged.add(onSumChanged);
+                }
+                await context.sync();
+            });
+        }
+
+        async function onSumChanged(eventArgs: Excel.BindingDataChangedEventArgs) {
+            await Excel.run(async (context) => {
+                // Log the address of the sum that changed
+                const range = context.workbook.bindings.getItem(eventArgs.binding.id).getRange();
+                const currentDate = new Date();
+                const cell = range.getCell(1, 0);
+                cell.formulas = [["=TIME(" + currentDate.getHours() + "," + currentDate.getMinutes() + "," + currentDate.getSeconds() + ")"]];
+                await context.sync();
+            });
+        }
+
+        async function setup() {
```

Comment:
I recommend that you have setup create a new sheet with some appropriate data. Then you don't have to worry about interfering with the user's sheets and the other methods can look for that sheet.